test/integration/targets/vmware_content_library/content_lib_crud_ops.yml

```diff
@@ -0,0 +1,73 @@
+# Test code for the vmware_content_library CRUD Operations.
+# Copyright: (c) 2019, Pavan Bidkar <pbidkar@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- when: vcsim is not defined
+  block:
+    - &content_lib_create
+      name: Create Content Library
+      vmware_content_library_manager:
+        hostname: '{{ vcenter_hostname }}'
+        username: '{{ vcenter_username }}'
+        password: '{{ vcenter_password }}'
+        validate_certs: False
+        library_name: Sample_Library
+        library_description: Sample Description
+        state: present
+      register: content_lib_create
+
```

Comment:
I get the following here here:

```
fatal: [testhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "datastore_name": null,
            "hostname": "vcenter.test",
            "library_description": "Sample Description",
            "library_name": "Sample_Library",
            "library_type": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "protocol": "https",
            "state": "present",
            "username": "administrator@vsphere.local",
            "validate_certs": false
        }
    },
    "msg": "Failed to find the datastore None"
}
```
To avoid the problem, I had to:

Load the `prepare_vmware_tests` role before:

```yaml
    - import_role:
        name: prepare_vmware_tests
      vars:
        setup_attach_host: true
        setup_datastore: true
```

and add the following parameter: `datastore_name: '{{ ds1 }}'`.

And then I get a KeyError exception in `state_create_library()`:
```
        create_spec.type = self.library_types[self.library_type]
```

The`library_type` parameter is `required=False`, maybe you should change it to `True`, or use a default value.

Reply:
Hi @goneri, Thanks for reporting this. actually datastore is required only for create Content library for other usecases (update & delete) datastore is not required. Same for library_type. Hence I made both params required=False. 
As you said, I should make them required. I will see how to handle this.